ThirdParty/DirectXShaderCompiler/dxc/WinAdapter.h

```diff
@@ -342,7 +342,9 @@
 #define __stdcall
 #define __vectorcall
 #define __thiscall
+#ifndef PLATFORM_FREEBSD
 #define __fastcall
```

Comment:
Is `__fastcall` a recognized keyword when compiling on FreeBSD?

Reply:
On this one, the compiler was complaining about fastcall being redefined, as FreeBSD's headers already define fastcall.

Reply:
Which compiler are you using?

Reply:
Believe it or not, the entirety of DiligentEngine was built using the compiler provided by the base system: clang 16.0.6.
![image](https://github.com/DiligentGraphics/DiligentCore/assets/965280/f88cabed-e057-4dc2-999b-015ab3a6743d)

Reply:
This is what happens without skipping `__fastcall`:

```
[ 35%] Building CXX object DiligentCore/Graphics/ShaderTools/CMakeFiles/Diligent-ShaderTools.dir/src/DXCompiler.cpp.o
In file included from /usr/home/jlhawkwell/Projects/DiligentEngine/DiligentCore/Graphics/ShaderTools/src/DXCompiler.cpp:42:
In file included from /usr/home/jlhawkwell/Projects/DiligentEngine/DiligentCore/Graphics/ShaderTools/include/DXCompilerBaseLinux.hpp:31:
In file included from /usr/home/jlhawkwell/Projects/DiligentEngine/DiligentCore/Graphics/ShaderTools/../../ThirdParty/DirectXShaderCompiler/dxc/dxcapi.h:37:
/usr/home/jlhawkwell/Projects/DiligentEngine/DiligentCore/Graphics/ShaderTools/../../ThirdParty/DirectXShaderCompiler/dxc/WinAdapter.h:345:9: error: '__fastcall' macro redefined [-Werror,-Wmacro-redefined]
#define __fastcall
        ^
/usr/include/sys/cdefs.h:365:9: note: previous definition is here
#define __fastcall      __attribute__((__fastcall__))
        ^
1 error generated.
gmake[2]: *** [DiligentCore/Graphics/ShaderTools/CMakeFiles/Diligent-ShaderTools.dir/build.make:122: DiligentCore/Graphics/ShaderTools/CMakeFiles/Diligent-ShaderTools.dir/src/DXCompiler.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3521: DiligentCore/Graphics/ShaderTools/CMakeFiles/Diligent-ShaderTools.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```

Reply:
I updated DXC headers, so this should work now - please try the latest version

Reply:
A better way to fix this is to check

`#ifndef __fastcall`

Though I do really not understand why it only complains about `__fastcall`. All other macros are defined in the same way.

Reply:
What I suggested is to use the `#ifndef __fastcall` guard instead of `#ifndef PLATFORM_FREEBSD`.
Also note that this file is a third-party header so with the next update, it may break again.